CHANGELOG.md

```diff
@@ -6,6 +6,10 @@
   * Use `applicationContext` in `DataCollector#collectDeviceData()` callback to prevent potential `Activity` leaks
 * GooglePay
   * Fix issue that causes `GooglePayNonce#isNetworkTokenized` to always return `false` after being parceled
+* **Potential Breaking Change**
```

Comment:
Shouldn't this be in a next major version? The Kotlin changes haven't gone out yet to V4, right?

Reply:
Yeah I agree that is should be next major version. There are Kotlin code changes out in a few of the latest releases. I'm doing some more investigating though it may be caused by a dependency on work-manager I'm hoping we can roll something back without having to worry about bumping the Java version.

Reply:
@scannillo This PR could fix it without us having to break: https://github.com/braintree/braintree_android/pull/708

Reply:
I feel like the most _SemVer_ route would be revert the existing Kotlin changes to v4, as a "bug fix" to the breaking changes. And then save Kotlin rewrites for v5, dropping Java 7.

Is the route you're thinking to go? Or were you leaning toward continuing to do Kotlin rewrites in V4?

Reply:
There may be a way. I went down the path of cherry picking the Java commits, but we do have a series of merge commits in the mainline that make reverting to Java 7 more complex.

The default compile target is now Java 8 in Android Studio so we should be good. In the worst case we will cut a separate pure Java 7 branch if we have a merchant that is unable to upgrade.

Reply:
Gotcha - I wonder if we can track Kotlin vs Java and version # in our analytics metadata in the future.

Reply:
I wish yeah the best way for us to get this information would be to survey our merchants AFAIK. Since this happens at compile time I don't think we can get access to the JDK version used to build a merchant app. By the time it gets to a consumer device it's all Java bytecode.